core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -143,25 +189,28 @@ class RecordInsightsLOCO[T <: Model[T]]
       val diffToExamine = computeDiffs(i, oldInd, oldVal, featureArray, featureSize, baseScore)
       val history = histories(oldInd)
 
-      // Let's check the indicator value and descriptor value
-      // If those values are empty, the field is likely to be a derived text feature (hashing tf output)
-      if (textFeatureIndices.contains(oldInd) && history.indicatorValue.isEmpty && history.descriptorValue.isEmpty) {
-        // Name of the field
-        val rawName = history.parentFeatureType match {
-          case s if s.exists(textMapTypes.contains) => {
-            val grouping = history.grouping
-            history.parentFeatureOrigins.headOption.map(_ + "_" + grouping.getOrElse(""))
+      // If indicator value and descriptor value values are empty, the field is likely
+      // to be a derived text feature (hashing tf output)
+      val isHashTextFeature = textFeatureIndices.contains(oldInd) &&
+        history.indicatorValue.isEmpty && history.descriptorValue.isEmpty
+      // If the descriptor value exists, the field is likely to be a derived date feature from unit circle transformer.
+      val isUnitCircleDateFeature = dateFeatureIndices.contains(oldInd) && history.descriptorValue.isDefined
+
+      if (isHashTextFeature || isUnitCircleDateFeature) {
+        for {name <- getRawFeatureName(history)} {
+          // Update the aggregation map for each (rawFeatureName, timePeriod) in case of date features.
+          val key = if (isUnitCircleDateFeature) {
+            name + "_" + history.descriptorValue.flatMap(convertToTimePeriod).map(_.entryName).getOrElse("")
```

Comment:
let's replace `isUnitCircleDateFeature` val  and `if / else`  with pattern matching

Reply:
pattern match on what exactly ? `history` ?

Reply:
btw, isUnitCircleDateFeature is referred multiple times, hence created a val.

Reply:
yes, you can probably pattern match on history and it should make the code more readable as you would have clearly separated cases. give it a try.

Reply:
and remember that you can pattern match together with `if` conditions.

Reply:
done.